tests/transforms/test_transpile.py

```diff
@@ -0,0 +1,57 @@
+"""
+Unit tests for transpiler function.
+"""
+from math import isclose
+import pytest
+
+from pennylane import numpy as np
+import pennylane as qml
+from pennylane.transforms.transpile import transpile
+
+
+def build_qfunc(wires):
+    def qfunc(x, y, z):
+        qml.Hadamard(wires=wires[0])
+        qml.RZ(z, wires=wires[2])
+        qml.CNOT(wires=[wires[2], wires[0]])
+        qml.CNOT(wires=[wires[1], wires[0]])
+        qml.RX(x, wires=wires[0])
+        qml.CNOT(wires=[wires[0], wires[2]])
+        qml.RZ(-z, wires=wires[2])
+        qml.RX(y, wires=wires[0])
+        qml.PauliY(wires=wires[2])
+        qml.CY(wires=[wires[1], wires[2]])
+        return qml.expval(qml.PauliZ(wires=wires[0]))
+
+    return qfunc
+
+
+class TestTranspile:
+    """ Unit tests for transpile function """
+
+    def test_transpile_invalid_coupling(self):
+        """ test that error is raised when coupling_map is invalid"""
+        dev = qml.device("default.qubit", wires=[0, 1, 2])
+
+        # build circuit
+        original_qfunc = build_qfunc([0, 1, 2])
+        transpiled_qfunc = transpile(coupling_map=[(0, 1)])(original_qfunc)
+        transpiled_qnode = qml.QNode(transpiled_qfunc, dev)
+        with pytest.raises(ValueError):
+            transpiled_qnode(0.1, 0.2, 0.3)
```

Comment:
It could be useful here to also use the `match=` argument for `pytest.raises`. This accepts as regex string that should match the error messaged raised, to ensure that the _correct_ `ValueError` is indeed raised.

Reply:
yep, I include this now